### ISSUE TITLE
Crash with Gitify

### ISSUE THREAD

**Issue by varlesh** [bilelmoussaoui/Hardcode-Tray#646]
config file:
https://github.com/bilelmoussaoui/Hardcode-Tray/blob/master/data/database/gitify.electron.json
application:
https://github.com/manosim/gitify/releases/download/v3.1.0/gitify_3.1.0_amd64.deb

```
alex@alex:~$ sudo -E hardcode-tray -ct RSVGConvert -o gitify.electron
Добро пожаловать в Hardcode-Tray!
Hardcode-Tray replaces some icons on the binary files of some applications. 
Please close all the running applications before running Hardcode-Tray script.
Рабочее окружение: Kde
Увеличение: 1
Размер значков: 22
Тема значков: Papirus-Dark
Средство конвертирования: RSVGConvert
Цель:Gitify.Electron
1 - Исправить
2 - Восстановить
3 - Очистить резервный кэш
Выберите:1
Исправление...


Traceback (most recent call last):
  File "/usr/bin/hardcode-tray", line 121, in <module>
    App.execute()
  File "/usr/lib/python3.6/site-packages/HardcodeTray/app.py", line 120, in execute
    delta = app.do_action(action)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/application.py", line 148, in do_action
    self.install()
  File "/usr/lib/python3.6/site-packages/HardcodeTray/decorators.py", line 58, in wrapper
    func(app)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/application.py", line 96, in install
    self.execute(Action.APPLY)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/helpers/binary.py", line 50, in execute
    self.install_icon(icon, icon_path)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/electron.py", line 41, in install_icon
    self.set_icon(icon, icon_path, png_bytes, True)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/electron.py", line 66, in set_icon
    asar.write(icon_to_replace, png_bytes)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/helpers/asar.py", line 106, in write
    new_files = change_dict_vals(self._header, sizediff, offset0)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 245, in change_dict_vals
    if d2.get('offset') and int(d2.get('offset')) > offset:
TypeError: int() argument must be a string, a bytes-like object or a number, not 'dict'
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 154, in apport_excepthook
    os.O_WRONLY | os.O_CREAT | os.O_EXCL, 0o640), 'wb') as f:
FileNotFoundError: [Errno 2] Нет такого файла или каталога: '/var/crash/_usr_bin_hardcode-tray.0.crash'

Original exception was:
Traceback (most recent call last):
  File "/usr/bin/hardcode-tray", line 121, in <module>
    App.execute()
  File "/usr/lib/python3.6/site-packages/HardcodeTray/app.py", line 120, in execute
    delta = app.do_action(action)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/application.py", line 148, in do_action
    self.install()
  File "/usr/lib/python3.6/site-packages/HardcodeTray/decorators.py", line 58, in wrapper
    func(app)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/application.py", line 96, in install
    self.execute(Action.APPLY)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/helpers/binary.py", line 50, in execute
    self.install_icon(icon, icon_path)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/electron.py", line 41, in install_icon
    self.set_icon(icon, icon_path, png_bytes, True)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/electron.py", line 66, in set_icon
    asar.write(icon_to_replace, png_bytes)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/modules/applications/helpers/asar.py", line 106, in write
    new_files = change_dict_vals(self._header, sizediff, offset0)
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in change_dict_vals
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 244, in <dictcomp>
    d2 = {k: change_dict_vals(v, sizediff, offset) for k, v in d.items()}
  File "/usr/lib/python3.6/site-packages/HardcodeTray/utils.py", line 245, in change_dict_vals
    if d2.get('offset') and int(d2.get('offset')) > offset:
TypeError: int() argument must be a string, a bytes-like object or a number, not 'dict'

```

**Comment by varlesh** [comment]
used Electron 9. Similar bug with https://github.com/bilelmoussaoui/Hardcode-Tray/issues/637

**Comment by varlesh** [comment]
LOG:
```
[DEBUG] 2020-06-27 10:00:24 DE: Kde
[DEBUG] 2020-06-27 10:00:24 Config file: Not found.
[DEBUG] 2020-06-27 10:00:24 Arguments/Icon Size: None
[DEBUG] 2020-06-27 10:00:24 System/Icon Size: 22
[DEBUG] 2020-06-27 10:00:24 System/Theme: Papirus-Dark
[DEBUG] 2020-06-27 10:00:24 Arguments/Conversion Tool: RSVGConvert
[DEBUG] 2020-06-27 10:00:24 Arguments/Only: gitify.electron
[DEBUG] 2020-06-27 10:00:27 Config/Backup Ignore: False
[DEBUG] 2020-06-27 10:00:27 Create new backup folder for Gitify
[DEBUG] 2020-06-27 10:00:27 Creating directory: /home/alex/.config/Hardcode-Tray/Gitify/2020-06-27 10:00:27/
[DEBUG] 2020-06-27 10:00:27 Executing command: rsvg-convert -f png -o /tmp/tmpdh0xy2ov -w 22 -h 22 /usr/share/icons/Papirus-Dark/22x22/panel/gitify-tray.svg
```

**Comment by acarasimon96** [comment]
Fixed via #711 and e12bf91 :tada:
